client/landing/stepper/declarative-flow/link-in-bio.ts

```diff
@@ -46,6 +47,13 @@ const linkInBio: Flow = {
 		];
 	},
 
+	useSideEffect() {
+		const { setHidePlansFeatureComparison } = useDispatch( ONBOARD_STORE );
+		useEffect( () => {
+			setHidePlansFeatureComparison( true );
+		} );
+	},
```

Comment:
I am not entirely sure, but I wouldn't object to going through the store (and haven't while reviewing). I think the onboard store is meant to serve things in a way that leads to less prop drilling. Otherwise, maybe most everything from that store can be passed around as props (probably not a great idea).

> Is there any use case where we might want to enable the comparison grid based on user action?

None of these are "user action" based. They are side effects to the root rendering.

Reply:
> None of these are "user action" based. They are side effects to the root rendering.

This action sets a value which is then fetched by the plans-wrapper. I was referring to these values, which are picked from the store:
https://github.com/Automattic/wp-calypso/blob/e7903abc2f7c3a2dd3102885df6602b49bd7db36/client/landing/stepper/declarative-flow/internals/steps-repository/plans/plans-wrapper.tsx#L63-L69

The values of `hideFreePlan` and `domainCartItem` depend on user action, but `hidePlansFeatureComparison` is a read-only value, which is a property of the flow itself. Typically something like this would be behind a `flowName` check, which IMO is flawed, but we don't have any better solutions yet. (pdDR7T-vO-p2)

> I think the onboard store is meant to serve things in a way that leads to less prop drilling.

It does, but I'm concerned if it would lead to the "dependency-soup" issue we have in the classic signup framework. Although, since Stepper flows are typed, it reduces that issue somewhat. 🤔

Reply:
> This action sets a value which is then fetched by the plans-wrapper. I was referring to these values, which are picked from the store:

Gotcha. It's a good point. It's almost like it becomes an implicit dependency on the flow right? (that it must set this value to get the grid showing). I don't know if it's a problem per se, but it does feel less of a "shared state" (at least across flows) than the rest. I think central dependency management wasn't a recipe for putting together Stepper, and I wouldn't disagree with that. 

> It does, but I'm concerned if it would lead to the "dependency-soup" issue we have in the classic signup framework. Although, since Stepper flows are typed, it reduces that issue somewhat. 🤔

There may be a bit of that risk, but with everything typed and within React, definitely nowhere near classic signup. The architecture seems solid without a ton of over-engineering, so it remains to be seen how well everything is served (hopefully soon, classic signup? `/start`).